jsonnet/components/alertmanager.libsonnet

```diff
@@ -13,6 +13,10 @@ function(params)
         name: 'alertmanager-trusted-ca-bundle',
         namespace: cfg.namespace,
         labels: {
+          'app.kubernetes.io/name': 'alertmanager-trusted-ca-bundle',
+          'app.kubernetes.io/managed-by': 'cluster-monitoring-operator',
+          'app.kubernetes.io/part-of': 'openshift-monitoring',
+          'app.kubernetes.io/component': 'configmap',
```

Comment:
Instead of redefining values for `component` and `name`, we should reuse the ones from upstream: `https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/alertmanager.libsonnet#L10-L15`. Generally the name shouldn't be the one of the resource but the name of the component so `alertmanager` in this case and the component should be the `role` of the component in the cluster so `alert-router` here.

What could be done to make this change more generic is to make the `app.kubernetes.io/part-of` and `app.kubernetes.io/part-of` labels configurable at the top level of kube-prometheus and then just apply the upstream commonLabels to the new resource that we introduce downstream.